crates/router/src/connector/aci/transformers.rs

```diff
@@ -148,176 +353,179 @@ pub enum AciPaymentType {
-                        merchant_transaction_id: None,
-                        customer_email: billing_details.email.clone(),
-                        shopper_result_url: item.request.router_return_url.clone(),
-                    })),
-
-                    api_models::payments::BankRedirectData::Interac { email, country } => {
-                        PaymentDetails::BankRedirect(Box::new(BankRedirectionPMData {
-                            payment_brand: PaymentBrand::InteracOnline,
-                            bank_account_country: Some(*country),
-                            bank_account_bank_name: None,
-                            bank_account_bic: None,
-                            bank_account_iban: None,
-                            billing_country: None,
-                            merchant_customer_id: None,
-                            merchant_transaction_id: None,
-                            customer_email: Some(email.to_owned()),
-                            shopper_result_url: item.request.router_return_url.clone(),
-                        }))
-                    }
-
-                    api_models::payments::BankRedirectData::Trustly { country } => {
-                        PaymentDetails::BankRedirect(Box::new(BankRedirectionPMData {
-                            payment_brand: PaymentBrand::Trustly,
-                            bank_account_country: None,
-                            bank_account_bank_name: None,
-                            bank_account_bic: None,
-                            bank_account_iban: None,
-                            billing_country: Some(*country),
-                            merchant_customer_id: Some(Secret::new(
-                                item.customer_id.clone().ok_or(
-                                    errors::ConnectorError::MissingRequiredField {
-                                        field_name: "customer_id",
-                                    },
-                                )?,
-                            )),
-                            merchant_transaction_id: Some(Secret::new(item.payment_id.clone())),
-                            customer_email: None,
-                            shopper_result_url: item.request.router_return_url.clone(),
-                        }))
-                    }
-
-                    _ => Err(errors::ConnectorError::NotImplemented(
-                        "Payment method".to_string(),
-                    ))?,
-                }
             }
+        } else if item.request.mandate_id.is_some() {
+            let instruction = get_instruction_details(item);
+            let txn_details = get_transaction_details(item)?;
+            let aci_payment_request = Self {
+                txn_details,
+                payment_method: PaymentDetails::Mandate,
+                instruction,
+                shopper_result_url: item.request.router_return_url.clone(),
+            };
+            Ok(aci_payment_request)
```

Comment:
Instead of having else if, lets move this inside your match statement in line 374. In case of non first mandate payments the item.request.payment_method_data will have `MandatePayment` value